doc/roadmap.rst

```diff
@@ -54,40 +62,44 @@ Architectural / general goals
 -----------------------------
 The list is numbered not as an indication of the order of priority, but to
 make referring to specific points easier. Please add new entries only at the
-bottom.
+bottom. Note that the crossed out entries are already done, and we try to keep
+the document up to date as we work on these issues.
 
-#. Everything in Scikit-learn should conform to our API contract
 
-   * `Pipeline <pipeline.Pipeline>` and `FeatureUnion` modify their input
-     parameters in fit. Fixing this requires making sure we have a good
-     grasp of their use cases to make sure all current functionality is
-     maintained. :issue:`8157` :issue:`7382`
-
-#. Improved handling of Pandas DataFrames and SparseDataFrames
+#. Improved handling of Pandas DataFrames
 
    * document current handling
    * column reordering issue :issue:`7242`
    * avoiding unnecessary conversion to ndarray :issue:`12147`
    * returning DataFrames from transformers :issue:`5523`
-   * getting DataFrames from dataset loaders :issue:`10733`, :issue:`13902`
+   * getting DataFrames from dataset loaders :issue:`10733`,
+     |ss| :issue:`13902` |se|
    * Sparse currently not considered :issue:`12800`
 
 #. Improved handling of categorical features
 
    * Tree-based models should be able to handle both continuous and categorical
-     features :issue:`4899`
-   * In dataset loaders :issue:`13902`
+     features :issue:`12866`
```

Comment:
we probably have separate issues for the old threes and gradient boosting, right?